tools/iceoryx_build_test.sh

```diff
@@ -258,23 +258,23 @@ fi
 if [ $CLEAN_BUILD == true ]
 then
     echo " [i] Cleaning build directory"
-    cd $WORKSPACE
-    rm -rf $BUILD_DIR/*
+    cd "$WORKSPACE"
+    rm -rf "${BUILD_DIR:?}/"*
```

Comment:
Does this check whether `BUILD_DIR` is empty?

Reply:
No it fails, if the variable $BUILD_DIR is not empty , but we set it always in this this script independent if the build directory exist or not.
I've added the `no-build` parameter to this script as you suggested here: https://github.com/eclipse-iceoryx/iceoryx/issues/920

Reply:
Please also link to #920 in the PR description